.github/workflows/setup-tofu.yml

```diff
@@ -0,0 +1,314 @@
+name: 'Setup OpenTofu'
```

Comment:
i have a lot of questions on this file, left couple but seems to have a lot of duplicates, i might not really understand this workflow

Reply:
It's the test pipelines definition copied over from the setup-tearraform repo. I didn't do thorough analysis of what bits we could remove. My focus was to deliver working github actions v0.0.1 so we could increment while enabling users to setup tofu in their CI. WDYT?

Reply:
I'm fine with this file now after your answers